resilience4j-spring-boot2/src/main/java/io/github/resilience4j/circuitbreaker/autoconfigure/CircuitBreakerConfigurationOnMissingBean.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2018 Slawomir Kowalski
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.github.resilience4j.circuitbreaker.autoconfigure;
+
+import io.github.resilience4j.circuitbreaker.CircuitBreakerRegistry;
+import io.github.resilience4j.circuitbreaker.configure.CircuitBreakerAspect;
+import io.github.resilience4j.circuitbreaker.configure.CircuitBreakerConfiguration;
+import io.github.resilience4j.circuitbreaker.configure.CircuitBreakerConfigurationProperties;
+import io.github.resilience4j.circuitbreaker.event.CircuitBreakerEvent;
+import io.github.resilience4j.consumer.EventConsumerRegistry;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class CircuitBreakerConfigurationOnMissingBean {
+
+    private final CircuitBreakerConfiguration circuitBreakerConfiguration;
+
+    CircuitBreakerConfigurationOnMissingBean() {
+        circuitBreakerConfiguration = new CircuitBreakerConfiguration();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public CircuitBreakerRegistry circuitBreakerRegistry(CircuitBreakerConfigurationProperties circuitBreakerProperties,
+                                                         EventConsumerRegistry<CircuitBreakerEvent> eventConsumerRegistry) {
+        return circuitBreakerConfiguration.circuitBreakerRegistry(circuitBreakerProperties, eventConsumerRegistry);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public CircuitBreakerAspect circuitBreakerAspect(CircuitBreakerConfigurationProperties circuitBreakerProperties,
+                                                     CircuitBreakerRegistry circuitBreakerRegistry) {
+        return circuitBreakerConfiguration.circuitBreakerAspect(circuitBreakerProperties, circuitBreakerRegistry);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
```

Comment:
This will not work how you think it will for a generic type bean because of erasure.  For this to work, we need to bump to spring boot 2.1.x where they added parameterizedContainer property to the @CondtionalOnMissingBean annotation.  That would look like this:
@ConditionalOnMissingBean(value = CircuitBreakerEvent.class, parameterizedContainer = EventConsumerRegistry.class)

I'm working on a change for resolving issue #298 and I needed the changes from this PR to solve that one so I pulled them into my fork.  I can fix these issues, but need to know what people think.  Can I upgrade the spring boot 2 library to use 2.1.x to fix this correctly, or do we want to just leave it so you can't override the EventConsumerRegistry for now?